hal/src/sercom/v2/uart/config.rs

```diff
@@ -249,12 +250,14 @@ where
     /// When set, the UART will detect collisions and update the
     /// corresponding flag in the STATUS register.
     #[inline]
+    #[cfg(not(feature = "samd20"))]
     pub fn set_collision_detection(&mut self, enabled: bool) {
         self.registers.set_collision_detection(enabled);
     }
 
     /// Get the current collision detector setting
     #[inline]
+    #[cfg(not(feature = "samd20"))]
```

Comment:
Maybe it would be a good idea to separate out all the functions unavailable to the SAMD20 out in their own `impl` block to avoid having to repeat the `#[cfg]s` at every function definition.